modules/infinite-scroll/infinity.php

```diff
@@ -449,7 +449,7 @@ function has_only_title_matching_posts() {
 		}
 
 		//actual testing. As search query combines multiple keywords with AND, it's enough to check if any of the keywords is present in the title
-		if ( false !== strpos( $post->post_title, current( $search_terms ) ) ) {
+		if ( ! empty( current( $search_terms ) ) && false !== strpos( $post->post_title, current( $search_terms ) ) ) {
```

Comment:
Prior to 5.5 you're not allowed to ass in arbitrary expressions into `empty()` 
http://php.net/manual/en/migration55.new-features.php#migration55.new-features.empty

maybe declare `current( $search_terms )` as it's own var before checking with empty?